### PR TITLE
Add trampoline functionality

### LINKED ISSUE #89: Add ability to write model classes in python

### PR DESCRIPTION
**Pull request by JamesYang007**
close #89 

Don't merge yet! It's not finished and requires a bit of reworking.

### CODE REVIEW

File: python/test/src/model/model.hpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <pybind11/pybind11.h>
```
[the remark this answers is not in the document]
Comment (JamesYang007) [reply python/test/src/model/model.hpp:2]: That seems a bit too aggressive - I'd be down to just put everything in `python/test/src/model.cpp`  and have a single header file `python/test/src/decl.hpp` that replaces `model.hpp`.

File: python/src/core.cpp [hunk 2]
```diff
@@ -19,5 +20,8 @@ PYBIND11_MODULE(core, m) {
 
     py::module_ bound_m = m.def_submodule("bound", "Bound submodule.");
     kevlar::bound::add_to_module(bound_m);
+
+    py::module_ std_m = m.def_submodule("std", "STD submodule.");
```
[the remark this answers is not in the document]
Comment (JamesYang007) [reply python/src/core.cpp:24]: Turns out, I don't end up needing this submodule lmao.

File: python/test/core/model/base_unittest.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def score(self, gridpt_idx, out):
+        out[...] = 2.1
+
+
+class PySGS(SimGlobalStateBase):
+    def make_sim_state(self):
+        return PySS()
+
+
+class TestBase(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (tbenthompson) [reply python/test/core/model/base_unittest.py:20]: AWESOME

File: python/test/src/model/model.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbenthompson) [reply python/test/src/model/model.hpp:2]: Sounds good.

**Review by JamesYang007 — COMMENTED** [review]
ready for merge now! teehee

**Pull request merged**